tests/test/wasm/test_wasm.cpp

```diff
@@ -81,106 +90,66 @@ TEST_CASE("Test binding", "[wasm]")
 {
+    cleanSystem();
+
     faabric::Message call;
     call.set_user("demo");
     call.set_function("x2");
 
-    module_cache::WasmModuleCache& registry =
-      module_cache::getWasmModuleCache();
-    wasm::WAVMWasmModule& cachedModule = registry.getCachedModule(call);
-
-    wasm::WAVMWasmModule module(cachedModule);
+    wasm::WAVMWasmModule module;
+    module.bindToFunction(call);
 
     // Perform first execution
     executeX2(module);
 
     // Reset
-    module = cachedModule;
+    module.reset(call);
 
     // Perform repeat executions on same module
     executeX2(module);
 
     // Reset
-    module = cachedModule;
+    module.reset(call);
 
     executeX2(module);
 }
 
 TEST_CASE("Test execution without binding fails", "[wasm]")
 {
-    faabric::Message callA;
-    callA.set_user("demo");
-    callA.set_function("dummy");
+    cleanSystem();
 
-    wasm::WAVMWasmModule module;
-    REQUIRE_THROWS(module.execute(callA));
-}
-
-TEST_CASE("Test binding twice fails", "[wasm]")
-{
     faabric::Message callA;
     callA.set_user("demo");
     callA.set_function("dummy");
 
     wasm::WAVMWasmModule module;
-    module.bindToFunction(callA);
-    REQUIRE_THROWS(module.bindToFunction(callA));
-}
-
-TEST_CASE("Test repeat execution with different function fails", "[wasm]")
-{
-    faabric::Message callA = faabric::util::messageFactory("demo", "dummy");
-    faabric::Message callB = faabric::util::messageFactory("demo", "x2");
-
-    wasm::WAVMWasmModule module;
-    module.bindToFunction(callA);
-
-    REQUIRE_THROWS(module.execute(callB));
+    REQUIRE_THROWS(module.executeFunction(callA));
 }
 
 TEST_CASE("Test reclaiming memory", "[wasm]")
 {
-    faabric::Message call = faabric::util::messageFactory("demo", "heap");
+    cleanSystem();
 
-    module_cache::WasmModuleCache& registry =
-      module_cache::getWasmModuleCache();
-    wasm::WAVMWasmModule& cachedModule = registry.getCachedModule(call);
+    faabric::Message call = faabric::util::messageFactory("demo", "heap");
 
-    wasm::WAVMWasmModule module(cachedModule);
+    wasm::WAVMWasmModule module;
+    module.bindToFunction(call);
 
     Uptr initialPages = Runtime::getMemoryNumPages(module.defaultMemory);
 
     // Run it (knowing memory will grow during execution)
-    module.execute(call);
+    module.executeFunction(call);
 
-    module = cachedModule;
+    module.reset(call);
 
     Uptr pagesAfter = Runtime::getMemoryNumPages(module.defaultMemory);
     REQUIRE(pagesAfter == initialPages);
 }
 
-TEST_CASE("Test GC", "[wasm]")
-{
-    wasm::WAVMWasmModule module;
-    faabric::Message call = faabric::util::messageFactory("demo", "malloc");
-
-    SECTION("Plain module")
-    {
-        // Do nothing
-    }
-    SECTION("Bound but not executed module") { module.bindToFunction(call); }
-    SECTION("Executed module")
-    {
-        module.bindToFunction(call);
-        module.execute(call);
-    }
-
-    bool success = module.tearDown();
-    REQUIRE(success);
-}
```

Comment:
WAVM GC will now fail an assertion if it fails. This should never happen unless we change the code in a way that makes it not possible to GC, in which case we should fix it.